### PR TITLE
feat: field autocomplete - sort labels alphabetically

### PR DESCRIPTION
**Pull request by IrakliJani**
Closes: https://github.com/lightdash/lightdash/issues/2773

### Description:

https://user-images.githubusercontent.com/962095/211298868-57cf84ba-f36d-4af5-843b-b397418b63ff.mp4

### CODE REVIEW

File: packages/frontend/src/components/common/Filters/FieldAutoComplete.tsx [hunk 1]
```diff
@@ -41,51 +42,59 @@ const FieldAutoComplete = <T extends Field | TableCalculation>({
...
-            }}
-        />
-    </>
-);
+}: FieldAutoCompleteProps<T>) => {
+    const sortedFields = useMemo(() => {
+        return fields.sort((a, b) =>
+            getItemLabel(a).localeCompare(getItemLabel(b)),
+        );
+    }, [fields]);
```
Comment (IrakliJani) [inline packages/frontend/src/components/common/Filters/FieldAutoComplete.tsx:50]: this is the only actual addition to this file

**Comment by ZeRego** [comment]
@TuringLovesDeathMetal is this something we want across the app? In the original ticket, they asked for the fields from the main table to be first and then the joined table fields.

**Comment by TuringLovesDeathMetal** [comment]
@ZeRego if we make it across teh app for now, it's better than what we have today 👌 

Ideally, the Explore View page would have a different behaviour to dashboard filters (since we have context on which is the main vs. joined tables). 

But if that's complicated, then I can open another ticket for it and we can do it separately :)

**Comment by IrakliJani** [comment]
@ZeRego good point.
I'm going to turn this to a draft PR and give it more though.
cc: @TuringLovesDeathMetal

**Comment by TuringLovesDeathMetal** [comment]
@IrakliJani honestly, I think it's fine as-is. 

We can always update it later if it's necessary, but having it just alphabetical is already a million times better than what it is now!

**Review by ZeRego — APPROVED** [review]

**Pull request merged**